src/dialects/maria/schema/columncompiler.js

```diff
@@ -0,0 +1,23 @@
+// MariaDB Column Compiler
+// -------
+import inherits from 'inherits';
+import ColumnCompiler_MySQL from '../../mysql/schema/columncompiler';
+
+import { assign } from 'lodash'
+
+function ColumnCompiler_MariaDB() {
+  ColumnCompiler_MySQL.apply(this, arguments);
+  this.modifiers = ['unsigned', 'nullable', 'defaultTo', 'first', 'after', 'comment', 'collate']
+}
+
+inherits(ColumnCompiler_MariaDB, ColumnCompiler_MySQL);
+
+assign(ColumnCompiler_MariaDB.prototype, {
+    // mariadb doesn't support json as of yet. 
+    // see https://jira.mariadb.org/browse/MDEV-9056 for issue status.
+  json () {
+    return 'text';
```

Comment:
this is here because maria doesn't have the json data type, and so needs to override the inherited definition of `json` with `text` so that table creation statements don't blow up.